Objective-C/CBLDatabase.mm

```diff
@@ -950,102 +952,105 @@ - (BOOL) saveDocument: (CBLDocument*)document
 #pragma mark - RESOLVING REPLICATED CONFLICTS:
 
 
-- (bool) resolveConflictInDocument: (NSString*)docID error: (NSError**)outError {
-    CBL_LOCK(self) {
-        C4Transaction t(_c4db);
-        if (!t.begin())
-            return convertError(t.error(), outError);
-        
-        // Read local document:
-        CBLDocument* localDoc = [[CBLDocument alloc] initWithDatabase: self
-                                                           documentID: docID
-                                                       includeDeleted: YES
-                                                                error: outError];
-        if (!localDoc)
-            return false;
+- (bool) resolveConflictInDocument: (NSString*)docID
+              withConflictResolver: (id<CBLConflictResolver>)conflictResolver
+                             error: (NSError**)outError {
+    while (true) {
+        CBLDocument* localDoc;
+        CBLDocument* remoteDoc;
+        CBL_LOCK(self) {
+            C4Transaction t(_c4db);
+            if (!t.begin())
+                return convertError(t.error(), outError);
+            
+            // Read local document:
+            localDoc = [[CBLDocument alloc] initWithDatabase: self documentID: docID
+                                              includeDeleted: YES error: outError];
+            if (!localDoc)
+                return false;
+            
+            // Read the conflicting remote revision:
+            remoteDoc = [[CBLDocument alloc] initWithDatabase: self documentID: docID
+                                               includeDeleted: YES error: outError];
+            if (!remoteDoc || ![remoteDoc selectConflictingRevision])
+                return false;
+            
+            if (!t.commit())
+                return convertError(t.error(), outError);
+        }
         
-        // Read the conflicting remote revision:
-        CBLDocument* remoteDoc = [[CBLDocument alloc] initWithDatabase: self
-                                                            documentID: docID
-                                                        includeDeleted: YES
-                                                                 error: outError];
-        if (!remoteDoc || ![remoteDoc selectConflictingRevision])
-            return false;
+        conflictResolver = conflictResolver ?: [CBLConflictResolution default];
+        CBLConflict* conflict = [[CBLConflict alloc] initWithLocalDocument: localDoc
+                                                            remoteDocument: remoteDoc];
```

Comment:
Need to check whether localDoc or remoteDoc is deleted doc or not. If it's a deleted doc, need to pass null. You may also add an assertion to make sure that both of them are not null.

Reply:
done